spec/controllers/tasks_controller_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+
+RSpec.describe TasksController, type: :controller do
+  let!(:user) { FactoryBot.create(:user) }
+  let!(:assignment) { FactoryBot.create(:assignment) }
+  let!(:task) { FactoryBot.create(:task, creator: user, user_ids: user.id) }
+  before { sign_in(user) }
+
+  describe 'GET #index' do
+    let(:task2) { FactoryBot.create(:task, creator: user, user_ids: user.id) }
+
+    it 'populates an array of all Tasks' do
+      get :index, params: { user_id: user.id }
+      expect(assigns(:tasks)).to match_array [task, task2]
+    end
+
+    it 'renders the :index template' do
+      get :index, params: { user_id: user.id }
+      expect(response).to render_template :index
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid attributes' do
```

Comment:
```
  describe 'POST #create' do
    context 'with valid attributes' do

      before { post :create, params: { user_id: user.id, task: task_attributes } }

      it 'creates a task' do
        expect(Task.count).to eq(1)
      end

      it 'redirects to the tasks index' do
        expect(response).to redirect_to user_tasks_path(user)
      end
    end
```

you can DRY your code with this befor...

Reply:
Thanks @tgi-wtag for the suggestion!

In addition to this, I would suggest another way for making it look more organized ⬇️ 

```
  describe 'POST #create' do
    context 'with valid attributes' do

      let(:create_action) { post :create, params: { task: task_attributes } }

      it 'creates a task' do
        expect { create_action }.to change(Task, :count).by(1)
      end

      it 'redirects to the tasks index' do
        expect(response).to redirect_to user_tasks_path(user)
      end
    end
```